shared/bazel/rules/objectivec_rules.bzl

```diff
@@ -0,0 +1,29 @@
+load("@rules_cc//cc:defs.bzl", "objc_library")
+
+OBJC_COMPILER_FLAGS = [
+    "-stdlib=libc++",
+    "-fPIC",
+]
+
+def wpilib_objc_library(
+        name,
+        srcs = [],
+        deps = [],
+        copts = None,
+        is_cpp = True,
+        include_arc = True,
+        **kwargs):
+    copts = copts or []
+    if is_cpp:
+        copts.append("-std=c++20")
+    if include_arc:
+        copts += ["-fobjc-weak", "-fobjc-arc"]
+
+    objc_library(
+        name = name,
+        srcs = srcs,
+        copts = copts + OBJC_COMPILER_FLAGS,
+        tags = ["manual"],  # This makes it so the other platforms will still build OK
```

Comment:
We can solve it in a future change if you want, but this is what `target_compatible_with = ["@platforms//os:osx"]` is for.